### DISCUSSION TITLE
Can we have cadl complier work in the browser?

### DISCUSSION THREAD

**Discussion by satishkumarnk-create** [microsoft/typespec#1490] [General]
We want to integrate the same cadl playground in our organization API help website. wanted to know if cadl complier can be triggered via javascript function and get open api specification doc in the browser. Point here is want to integrated same cadl playground in our website.

**Comment by NielsCo** [comment]
As the entire project is written in TypeScript and open-source I don't think this should be a problem.
I haven't tried it yet but you should be able to just use the npm packages and include them into a web-build.
I think looking at the tests of the openAPI3 emitter to see which methods call the actual code would be a good starting point

**Comment by markcowl** [comment]
There is a backlog work item for releasing the playground as a consumable package.  You can track progress / add comments here: https://github.com/microsoft/typespec/issues/1842
